### ISSUE TITLE
Reset console palette to default

### ISSUE THREAD

**Issue by gentoo90** [tomakita/Colorful.Console#16]
Changes to the console palette made by `Colorful.Console` remain after application exit.
This messes up colors in other console programs. Like FAR Manager:
- before running `ExampleConsoleApplication`:
  ![farbefore](https://cloud.githubusercontent.com/assets/1297574/19694611/ba0256c2-9ae8-11e6-956a-c58d2ac0dff3.png)
- and after running it:
  ![farafter](https://cloud.githubusercontent.com/assets/1297574/19694624/c62afc42-9ae8-11e6-9a78-06cb7d925b7a.png)

**Comment by tomakita** [comment]
Interesting.  Thanks for reporting this -- I'll look into this over the weekend, and will update this thread at that time.

**Comment by tomakita** [comment]
After looking into this and doing some testing, we can implement this feature, but only partially.

The Windows console can only display 16 different colors in a given session (where a session lasts from console open to console close).  So in the following scenario, the console would not be able to have its defaults restored:

1. Open a console
2. Load your application that uses Colorful.Console
3. Use 16 different non-standard (i.e. non-System.ConsoleColor) colors
4. Exit your application that uses Colorful.Console, but keep the console open

Since 16 different colors have already been used in the session, resetting the console's colors back to their defaults will have no visible effect.  On the other hand, in the following scenario, the console would be able to have _some_ of its defaults restored:

1. Open a console
2. Load your application that uses Colorful.Console
3. Use c < 16 different non-standard (i.e. non-System.ConsoleColor) colors
4. Exit your application that uses Colorful.Console, but keep the console open

If we were to implement a "reset to defaults" feature, the console would now have access to 16 - c default colors.  It might be tricky to allow the user to specify _which_ of the default colors would be accessible, though.

Does that make sense?  If so, would it still be useful to have a "reset to defaults" feature?

**Comment by gentoo90** [comment]
Maybe it would make more sense to just use default colors without changing them, which is currently impossible. Maybe additional `Write*` methods with `ConsoleColor` as the last argument?
Currently when I write
```C#
Console.Write("Qwerty", Color.Red)
```
it will not just use `ConsoleColor.Red`, but will tie `Color.Red` to the first available cell.
To workaround this I started `Main` with:
```C#
Console.Write("", Color.Navy);
Console.Write("", Color.Green);
Console.Write("", Color.Teal);
Console.Write("", Color.Maroon);
Console.Write("", Color.Purple);
Console.Write("", Color.Olive);
Console.Write("", Color.Silver);
Console.Write("", Color.Gray);
Console.Write("", Color.Blue);
Console.Write("", Color.Lime);
Console.Write("", Color.Cyan);
Console.Write("", Color.Red);
Console.Write("", Color.Fuchsia);
Console.Write("", Color.Yellow)
```
which ties `System.Drawing` colors to the corresponding console colors. But this doesn't cover the case when a user has non-standard colors in his console settings.

**Comment by tomakita** [comment]
I think I've just figured out how to do this.  I had misunderstood that way that the Win32 API keeps track of color changes -- rather, I thought that it kept track of the number of times that color had been changed, but it does not.  This means that I only need to make a small addition to the Colorful.Console API in order to support resetting colors back to defaults.  This would look something like:

Console.ClearStyle();

This would reset the color palette to the 16 default ConsoleColors.  The only caveat is that all text written to the console prior to the call to ClearStyle() would change color back to the default palette colors.  I know it's been a while since you opened this issue, so you may not be interested in this, but I figured it was a good feature to add anyway.  Thanks again for opening the issue!

**Comment by tomakita** [comment]
This has been fixed in [1.0.6](https://github.com/tomakita/Colorful.Console/tree/1.0.6)  If this isn't what you had in mind, please let me know!
